From 8e6678a00f81fffcd8a3bf71947066db1fffbd30 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Thu, 4 Dec 2014 23:16:54 +0000
Subject: [PATCH] MythDownloadManager: unlock m_infoLock before continuing
 after an error condition

Problem spotted by nouglywires on the forums.
---
 mythtv/libs/libmythbase/mythdownloadmanager.cpp |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/mythtv/libs/libmythbase/mythdownloadmanager.cpp b/mythtv/libs/libmythbase/mythdownloadmanager.cpp
index faac3ad..0902d11 100644
--- a/mythtv/libs/libmythbase/mythdownloadmanager.cpp
+++ b/mythtv/libs/libmythbase/mythdownloadmanager.cpp
@@ -312,7 +312,10 @@ void MythDownloadManager::run(void)
             m_downloadQueue.pop_front();
 
             if (!dlInfo)
+            {
+                m_infoLock->unlock();
                 continue;
+            }
 
             QUrl qurl(dlInfo->m_url);
             if (m_downloadInfos.contains(qurl.toString()))
-- 
1.7.10.2

